### DISCUSSION TITLE
Accuracy Metric

### DISCUSSION THREAD

**Discussion by jesamir** [Project-MONAI/MONAILabel#1715] [Q&A] [answered]
Hi everyone!

I have a quick question about the Accuracy metric GUI in the MONAI Label extension on 3D Slicer. Based on previous discussions, there's an automatic 20% split from training for validation. If, for example, I've manually labeled 30 images using segmentation.py (out of a total of unlabeled 300 images) and trained on each one, what does the accuracy metric of 60% in the GUI signify? Would it mean it is using that trained and labeled image for validation?

Thanks!

**Comment by diazandr3s** [comment]
Hi @jesamir,

That's right. By default, MONAI Label uses 20% of the training set to get the dice score you see in the GUI. 

You could [redefine this method](https://github.com/Project-MONAI/MONAILabel/blob/fc9b5676edbce419fe963f5a4a5361ce5bc2fb82/monailabel/tasks/train/basic_train.py#L398-L416) in the trainers file to use a different validation set or [change the percentage](https://github.com/Project-MONAI/MONAILabel/blob/fc9b5676edbce419fe963f5a4a5361ce5bc2fb82/monailabel/tasks/train/basic_train.py#L163)

Bear in mind that you'll need an annotated dataset to validate/test your model. Dice score is a measure of the similarity between the prediction and ground truth.

Hope this helps,

**Reply by jesamir** [reply]
Thank you so much! For validation and testing the model, do I press on the Submit Label GUI only while with training, press on the Submit Label GUI and the Train? A follow-up question: if I do end up pressing the Submit Label GUI while also pressing train, what would then the accuracy represent?

**Reply by diazandr3s** [reply, marked as answer]
Hi @jesamir,

I realise the terms testing and validation here may cause confusion. MONAI Label uses 20% of the annotated dataset to get a Dice score so user get a sense of how the model is doing. It is up to the user and the use case how they interpret this score.

When you hit Submit Label using the GUI, MONAI Label adds that annotation to the annotated set. 
When you hit Train, MONAI Lable splits the annotated dataset into 80/20 for training an getting the Dice score. 

> A follow-up question: if I do end up pressing the Submit Label GUI while also pressing train, what would then the accuracy represent?

If you hit Submit Label and then Train, MONAI Label adds the annotation to the labelled dataset and then splits 80/20 the dataset to train or retrain the model.

The score you see is always the one obtained from previous epoch of training.

Hope this helps,

**Reply by jesamir** [reply]
This makes sense now! Thank you so much for your help!
